js/ui/windowManager.js

```diff
@@ -2230,13 +2254,16 @@ const CodingManager = new Lang.Class({
         dst.show();
         dst.opacity = 0;
 
-        let dst_geometry = src.get_meta_window().get_frame_rect();
-        dst.get_meta_window().move_resize_frame(false,
-                                                dst_geometry.x,
-                                                dst_geometry.y,
-                                                dst_geometry.width,
-                                                dst_geometry.height);
-
+        let src_geometry = src.meta_window.get_frame_rect();
+        let isMaximized = (dst.meta_window.maximized_horizontally &&
+                           dst.meta_window.maximized_vertically);
+        if (isMaximized)
+            dst.meta_window.unmaximize(Meta.MaximizeFlags.BOTH);
```

Comment:
You don't want to unconditionally maximise here - what happens if the src window is also maximised?